Refactored development docs (#1983)
foundry-rs/starknet-foundry: CONTRIBUTING.md

# Contribution Guideline

Starknet Foundry is under active development and is open for contributions!

## Opening an issue 

If you think something doesn't work or something is missing please open an issue! This way we can address this problem

> and make Starknet Foundry better!

Reviewer: why the exclamation marks

Author: To be enthusiastic!

> Before opening an issue it is always a good idea to search existing 

Reviewer 2: ```suggestion
Before opening an issue, it is always a good idea to search existing 
```

> [issues](https://github.com/foundry-rs/starknet-foundry/issues) and verify if similar one doesn't already exist. 

Reviewer 2: ```suggestion
[issues](https://github.com/foundry-rs/starknet-foundry/issues) and verify if a similar one doesn't already exist. 
```

## Opening a pull request

> ### Environment setup

Reviewer: That's not a part of opening a pull request

Author: I can name this section differently, `Contributing` i think

See [development guide](https://foundry-rs.github.io/starknet-foundry/development/environment-setup.html) in Starknet
Foundry book for environment setup.

> ### Selecting an issue

Reviewer: You select an issue before you open PR

Author: This is a name of the section and this a first subsection

Author: I will rename the parent section

Author: Done

> If you are a first time contributor pick up any issue labeled as `good-first-issue`! Write a comment that you would like to 
> work on it and we will assign it to you. Need some guidance? Reach out to other developers on [Telegram](https://t.me/+d8ULaPxeRqlhMDNk) or open
> a [GitHub discussion](https://github.com/foundry-rs/starknet-foundry/discussions)!

Reviewer: ```suggestion
If you are a first time contributor, feel free to pick up any issue labeled as `good-first-issue`. Write a comment that you would like to 
work on it and we will assign it to you. Need some guidance? Reach out to other developers on [Telegram](https://t.me/starknet_foundry).
```
We don't use discussions, come on :P

Author: I mean somebody could use that, I would prefer to have another option mentioned if somebody does not want to use telegram for some reason. I don't see any cost of mentioning this that is why I decided to leave it.

Reviewer: also, excessive exclamation marks

Author: You should be more enthusiastic!

Author: I will drop the discussions info then.

Author: Done

> If you are more experienced Starknet Foundry contributor you can pick any issue labeled as `help-wanted`. Make sure to discuss the details with the core team beforehand.

Reviewer 2: ```suggestion
If you are a more experienced Starknet Foundry contributor you can pick any issue labeled as `help-wanted`. Make sure to discuss the details with the core team beforehand.
```

> Are you an experienced Starknet Foundry contributor you can pick any issue labeled as `help-wanted`. Make sure to discuss the details with the core team beforehand!

Reviewer: ```suggestion
If you are more experienced Starknet Foundry contributor you can pick any issue labeled as `help-wanted`. Make sure to discuss the details with the core team beforehand.
```

### Writing Tests

Please make sure the feature you are implementing is thoroughly tested with automatic tests.
You can check existing tests in the repository to see the recommended approach to testing.

### Pull Request Size

Try to make your pull request self-contained, only introducing the necessary changes.
If your feature is complicated,
consider splitting the changes into meaningful parts and introducing them as separate pull requests.

> Creating very large pull requests may significantly increase review time or event prevent them from being merged.

Reviewer 2: ```suggestion
Creating very large pull requests may significantly increase review time or even prevent them from being merged.
```
